# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../mesa/compile.patch.cross
# Copyright (C) 2010 - 2011 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- Mesa-7.8.2/src/glsl/Makefile.template.orig	2010-04-27 21:41:21.000000000 +0000
+++ Mesa-7.8.2/src/glsl/Makefile.template	2010-08-13 10:06:42.223158022 +0000
@@ -14,6 +14,7 @@
 ### Basic defines ###
 
 OBJECTS = $(C_SOURCES:.c=.o)
+OBJX = $(C_SOURCES:.c=.xo)
 
 INCLUDES = \
 	-I. \
@@ -22,11 +23,14 @@
 
 ##### TARGETS #####
 
-default: depend lib$(LIBNAME).a
+default: depend lib$(LIBNAME).a lib$(LIBNAME)x.a
 
 lib$(LIBNAME).a: $(OBJECTS) Makefile $(TOP)/src/glsl/Makefile.template
 	$(MKLIB) -o $(LIBNAME) -static $(OBJECTS)
 
+lib$(LIBNAME)x.a: $(OBJX) Makefile $(TOP)/src/glsl/Makefile.template
+	$(MKLIB) -o $(LIBNAME)x -static $(OBJX)
+
 depend: $(C_SOURCES)
 	rm -f depend
 	touch depend
@@ -35,6 +39,7 @@
 # Remove .o and backup files
 clean:
 	rm -f $(OBJECTS) lib$(LIBNAME).a depend depend.bak
+	rm -f $(OBJX) lib$(LIBNAME)x.a
 
 # Dummy target
 install:
@@ -46,5 +51,8 @@
 .c.o:
 	$(CC) -c $(INCLUDES) $(CFLAGS) $(LIBRARY_DEFINES) $< -o $@
 
+%.xo: %.c
+	$(HOSTCC) -c $(INCLUDES) $(CFLAGS) $(LIBRARY_DEFINES) $< -o $@
+
 -include depend
 
